generator/internal/discovery_type_vertex.cc

```diff
@@ -230,6 +258,32 @@ DiscoveryTypeVertex::DetermineReservedAndMaxFieldNumbers(
   return message_properties;
 }
 
+Status DiscoveryTypeVertex::UpdateTypeNames(
```

Comment:
nit: I cannot see why this is a class member vs. a standalone function. But surely the compiler would have yelled at you to make it `static`. I might just be blind.